### DISCUSSION TITLE
Uncaught TypeError: Cannot read properties of undefined (reading 'uri')

### DISCUSSION THREAD

**Discussion by jochemgruter** [livewire/livewire#6993] [1. Help] [answered]
I just upgraded to livewire v3 and on every page in my application I got the following error:

       Uncaught TypeError: Cannot read properties of undefined (reading 'uri')

I've auto inject the assets enabled. Eveything else is working fine, but I can't figure out why it's displaying this error in the console.

**Comment by g-gullstrand** [comment]
Got the same when using the "@livewireScriptConfig" directive. (Manually bundling Livewire and Alpine).  Changing to "@livewireScripts" and relying on auto injection fixes it. Not sure why but but it seems "window.livewireScriptConfig" is undefined

Console shows error on this line:
`var updateUri = (_ref107 = (_document$querySelect = (_document$querySelect2 = document.querySelector("[data-uri]")) === null || _document$querySelect2 === void 0 ? void 0 : _document$querySelect2.getAttribute("data-uri")) !== null && _document$querySelect !== void 0 ? _document$querySelect : window.livewireScriptConfig["uri"]) !== null && _ref107 !== void 0 ? _ref107 : null;`

**Reply by hugopozos** [reply]
Do you find another solution?

**Comment by georgeorl** [comment]
I got this error. Wrote [my solution](https://stackoverflow.com/questions/77519540/laravel-livewire-uncaught-typeerror-cannot-read-properties-of-undefined-readi/77521617#77521617) at stackoverflow.

[This PR](https://github.com/livewire/livewire/pull/6891) can prevent issues like that, but they doesn't want to accept PR.

**Comment by snapey** [comment]
I have this same issue after following @PhiloNL talk at LaraconEU 2024

from his live-coding, I created a general purpose modal component that uses @entangle to share its `modalState` with the modal and alpine and uses dynamic-component within the modal view to load the modal content (another livewire component)

This dynamically loaded content has an input box and as soon as anything is sent from this input, the console shows an error and nothing is sent to the backend.

```
livewire.js?id=6b5eb707:424 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'uri')
    at getUpdateUri (livewire.js?id=6b5eb707:424:119)
    at sendRequest (livewire.js?id=6b5eb707:4187:21)
    at RequestPool.send (livewire.js?id=6b5eb707:3964:13)
    at livewire.js?id=6b5eb707:4109:14
    at Set.forEach (<anonymous>)
    at CommitBus.createAndSendNewPool (livewire.js?id=6b5eb707:4105:13)
    at livewire.js?id=6b5eb707:4081:16
    at livewire.js?id=6b5eb707:4144:7
```
The line it fails at
```
  function getUpdateUri() {
    return document.querySelector("[data-update-uri]")?.getAttribute("data-update-uri") ?? window.livewireScriptConfig["uri"] ?? null;
  }
```

The odd thing is that other livewire components work fine, and even the dynamically included component works ok when placed directly on the page and not inside the modal.

**Reply by PhiloNL** [reply]
Can you verify that Livewire is loaded correctly? If you open the browser source, can you find the script tag for `livewire.js` and does it include the `data-uri` attribute?

**Reply by snapey** [reply]
Injected at the very bottom of the page;
```
<!-- Livewire Scripts -->
<script src="[/livewire/livewire.js?id=6b5eb707](http://speakernet.test/livewire/livewire.js?id=6b5eb707)"   data-csrf="ad6DeKCZ7UWkRQi5fwjcApJtEuVrw6z9g6tnT7MH" data-update-uri="/livewire/update" data-navigate-once="true"></script>
</body>
</html>

**Reply by snapey** [reply]
Running the line that breaks, directly in the console;
```
document.querySelector("[data-update-uri]")?.getAttribute("data-update-uri") ?? window.livewireScriptConfig["uri"] ?? null;
```
returns correctly `'/livewire/update'`

**Reply by PhiloNL** [reply]
Are you able to provide a reproducible demo? Preferably a very simple setup on https://wirebox.app/

**Reply by snapey** [reply]
I'm struggling with things like adapting to lack of blade components folder

**Reply by snapey** [reply]
Can't make this react to events so I'm giving up. https://wirebox.app/b/yrdkk 
Spent a day on this and I'm getting nowhere where a simple component with modal behaviour works perfectly.

**Reply by joshhanley** [reply]
@snapey hmm yeah that seems strange it's failing over and trying to load the script config option instead of the url off the normal script tag.

I know you've probably spent way to much time on it, but if you do decide you want to work out what is happening, can you make a small example repo (if that's easier than wirebox)? I will have a look to see what is going on.

**Reply by snapey** [reply]
I stripped the existing project right down and placed in public repo https://github.com/snapey/livewire3uri

No database is required only composer install and create .env  

Please note that the project was recently upgraded from Livewire 2 to Livewire 3 and the automatic upgrade tool was used

**Reply by joshhanley** [reply]
@snapey thanks! ~~What route/ components should I be looking at?~~

Edit: Don't worry worked it out

**Reply by joshhanley** [reply]
@snapey turns out that you were missing a closing `</div>` in `resources/views/components/modal.blade.php`.

Because of that when the component was diffing, it was removing the Livewire script tag due to the incorrectly formatted html, hence the error was being thrown.

<img width="919" alt="image" src="https://github.com/livewire/livewire/assets/882837/7e3ace4b-5109-472c-9f28-6f92eeaf3c00">

**Reply by snapey** [reply]
THANKS Josh for looking into this.  I'm not sure how i feel about it. I annoyed at my clumsiness but also the situation that Livewire got itself into with no real hint at the issue. I guess it didn't help that Livewire.js was inserted right after the modal with this also being added to the bottom of the master layout.  Anyway, thanks to you and @PhiloNL . At least this discussion might help anyone having the same issue.

**Reply by joshhanley** [reply]
@snapey no worries! 🙂 yeah agreed about the weird feeling, I feel like we could probably add a better response here.

Even just an console warning saying something like "scripts/scriptConfig couldn't be found - check your views to ensure everything is rendering correctly".

**Reply by PhiloNL** [reply, marked as answer]
Added a PR (https://github.com/livewire/livewire/pull/8042 ) which will add more context to the error and what the cause is likely to be :)
